contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/insertion/DefaultInsertionCostCalculator.java

```diff
@@ -55,6 +60,23 @@ public double calculate(DrtRequest drtRequest, Insertion insertion, DetourTimeIn
 			return INFEASIBLE_SOLUTION_COST;
 		}
 
+		// divert right now
+		if(insertion.pickup.index == 0) {
+			Waypoint nextWaypoint = insertion.pickup.nextWaypoint;
+			// there is an existing stop following the new insertion
+			if(nextWaypoint instanceof Waypoint.Stop && nextWaypoint != insertion.pickup.newWaypoint) {
+				// passengers are being dropped off == may be close to arrival
+				if(!((Waypoint.Stop) nextWaypoint).task.getDropoffRequests().isEmpty()) {
```

Comment:
What if the second next waypoint is dropping off and is within the defined time limit?

Reply:
That's a good point. This means we need to iterate over all scheduled stops!

Reply:
... all scheduled stops within the `fixedApproachTime` horizon

Reply:
Hi @michalmac 
I updated the code to check for all scheduled stops and check whether the actual detour caused is >0 (in some configurations it could also be that the new insertion is after the next stop and only adds an additional passenger to an existing stop without increasing stop duration, in which case there wouldn't be any additional detour)